### ISSUE TITLE
Async validation?

### ISSUE THREAD

**Issue by drewhamlett** [vuejs/vue-validator#10]
Using Promises?

```js
    validator: {
      validates: {
        email: function (val) {
          return /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(val)
        },
        remote: function (val) {
          new Promise(function (resolve, reject) {
            setTimeout(function () {
              if (val.length > 3) {
                return resolve(true)
              }
              return reject();
            }, 500);
          });
        }
      }
    },
```

**Comment by kazupon** [comment]
Sorry, Not support the Promise in vue-validator.
I would like to think whether become able to use the Promise in vue-validator.

**Comment by kazupon** [comment]
I published v1.4.0 release.
Please try it !!

**Comment by drewhamlett** [comment]
Thank you!
